### PR TITLE
[hailctl] Use python3 -m pip for pip

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/Makefile [hunk 1]
```diff
@@ -12,6 +12,7 @@ SPARK_VERSION = 2.4.0
 HAIL_PIP_VERSION = 0.2.14
 
 HAIL_PYTHON3 ?= python3
+pip = python3 -m pip
```
Comment (cseed) [inline hail/Makefile]: It is traditional to use upper-case names for make variables.   I would vote for `PIP`.

Comment (tpoterba) [reply hail/Makefile]: From [GNU make docs](https://www.gnu.org/software/make/manual/html_node/Using-Variables.html):

> It is traditional to use upper case letters in variable names, but we recommend using lower case letters for variable names that serve internal purposes in the makefile, and reserving upper case for parameters that control implicit rules or for parameters that the user should override with command options

I think I consider this in the latter category. Though now I see this should be a `:=` and should use HAIL_PYTHON3.

Comment (danking) [reply hail/Makefile]: I'm with Tim and the make docs, fwiw.

Comment (danking) [reply hail/Makefile]: but the inconsistency is jarring.

Comment (tpoterba) [reply hail/Makefile]: I think we should try to be consistent and organized -- maybe separate the things we view as user parameters and put those up top in caps?

**Pull request merged**